pyprotista/parsers/ident/comet_2020_01_4_parser.py

```diff
@@ -110,69 +52,177 @@ def check_parser_compatibility(cls, file):
+            entry.clear()
+
+    def get_peptide_lookup(self):
+        """Retrieve peptide ids with their corresponding sequences and modifications from xml.
+
+        Returns:
+            peptide_lookup (dict): peptide id with corresponding modifications and sequence
+        """
+        peptide_lookup = {}
+
+        modifications = ""
+        sequence = {}
+        peptide_information = False
+
+        for event, entry in etree.iterparse(self.input_file):
+            entry_tag = entry.tag
+
+            if entry_tag.endswith("DBSequence"):
+                peptide_information = True
+            elif peptide_information is True:
+                if entry_tag.endswith("PeptideSequence"):
+                    sequence = entry.text
+                    if len(self.fixed_mods) > 0:
+                        sequence_mod_map = self.map_mods_sequences(sequence)
+                        modifications = sequence_mod_map + ";"
+                    else:
+                        modifications = ""
+                elif entry_tag.endswith("Modification"):
+                    mass = entry.attrib["monoisotopicMassDelta"]
+                    location = entry.attrib["location"]
+                    mass_name = self.mod_mass_map[mass]
+                    modifications += mass_name + ":" + location + ";"
+                elif entry_tag.endswith("Peptide"):
+                    modifications = modifications.rstrip(";").lstrip(";")
+                    peptide_lookup[entry.attrib["id"]] = (modifications, sequence)
+                elif entry_tag.endswith("PeptideEvidence"):
+                    return peptide_lookup
+            entry.clear()
+
+    def get_spec_records(self):
+        """Retrieve specs from file.
+
+        Returns:
+            spec_records (list): information on PSMs
+        """
+        spec_records = []
+
+        spec_results = {}
+        spec_ident_items = []
+        spec_information = False
+
+        for event, entry in etree.iterparse(self.input_file):
+            entry_tag = entry.tag
+
+            if entry_tag.endswith("Inputs"):
+                spec_information = True
+            elif spec_information is True:
+                if entry_tag.endswith("cvParam"):
+                    if entry.attrib["name"] in self.mapping_dict:
+                        spec_results.update(
+                            {
+                                self.mapping_dict[entry.attrib["name"]]: entry.attrib[
+                                    "value"
+                                ]
+                            }
+                        )
+                elif entry_tag.endswith("SpectrumIdentificationItem"):
+                    for attribute in list(entry.attrib):
+                        if attribute in self.mapping_dict.keys():
+                            spec_results.update(
+                                {self.mapping_dict[attribute]: entry.attrib[attribute]}
+                            )
```

Comment:
see above
something like that maybe:
``` 
_key = self.mapping_dict[attribute]
spec_results[_key] = entry.attrib[attribute]
```

Reply:
I changed all update calls where this was possible to exactly this solution :)